src/test/java/com/dynatrace/openkit/protocol/StatusResponseTest.java

```diff
@@ -16,74 +16,402 @@
+        // given
+        expectedException.expect(NumberFormatException.class);
+
+        // when, then
+        new StatusResponse(StatusResponse.RESPONSE_KEY_SEND_INTERVAL + "=", 200);
+    }
+
+    @Test
+    public void parsingNonNumericSendIntervalValueThrowsException() {
+        // given
+        expectedException.expect(NumberFormatException.class);
+
+        // when, then
+        new StatusResponse(StatusResponse.RESPONSE_KEY_SEND_INTERVAL + "=a", 200);
+    }
+
     @Test
-    public void parseValidResponse() {
+    public void parsingTooBigSendIntervalValueThrowsException() {
         // given
-        String response = "type=m&cp=1&si=2&bn=someName&id=3&bl=4&er=1&cr=1";
-        int responseCode = 418;
+        expectedException.expect(NumberFormatException.class);
+
+        // when parsing 2^31, then
+        new StatusResponse(StatusResponse.RESPONSE_KEY_SEND_INTERVAL + "=2147483648", 200);
+    }
+
+    @Test
+    public void parsingMonitorNames() {
+        // when it's a positive number, then
+        assertThat(new StatusResponse(StatusResponse.RESPONSE_KEY_MONITOR_NAME + "=", 200).getMonitorName(), isEmptyString());
+        assertThat(new StatusResponse(StatusResponse.RESPONSE_KEY_MONITOR_NAME + "=foobar", 200).getMonitorName(), is(equalTo("foobar")));
+        assertThat(new StatusResponse(StatusResponse.RESPONSE_KEY_MONITOR_NAME + "=1234", 200).getMonitorName(), is(equalTo("1234")));
+    }
 
-        // when
-        StatusResponse statusResponse = new StatusResponse(response, responseCode);
+    @Test
+    public void serverIDIsParsed() {
+        // given
+        StatusResponse target = new StatusResponse(StatusResponse.RESPONSE_KEY_SERVER_ID + "=1234", 200);
 
         // then
-        assertThat(statusResponse.getResponseCode(), is(responseCode));
-        assertThat(statusResponse.isCapture(), is(true));
-        assertThat(statusResponse.getSendInterval(), is(2 * 1000));
-        assertThat(statusResponse.getMonitorName(), is("someName"));
-        assertThat(statusResponse.getServerID(), is(3));
-        assertThat(statusResponse.getMaxBeaconSize(), is(4 * 1024));
-        assertThat(statusResponse.isCaptureErrors(), is(true));
-        assertThat(statusResponse.isCaptureCrashes(), is(true));
+        assertThat(target.getServerID(), is(equalTo(1234)));
+    }
+
+    @Test
+    public void parsingEmptyServerIDValueThrowsException() {
+        // given
+        expectedException.expect(NumberFormatException.class);
+
+        // when, then
+        new StatusResponse(StatusResponse.RESPONSE_KEY_SERVER_ID + "=", 200);
+    }
+
+    @Test
+    public void parsingNonNumericServerIDValueThrowsException() {
+        // given
+        expectedException.expect(NumberFormatException.class);
+
+        // when, then
+        new StatusResponse(StatusResponse.RESPONSE_KEY_SERVER_ID + "=a", 200);
+    }
+
+    @Test
+    public void parsingTooBigServerIDValueThrowsException() {

```

Comment:
what about negative server IDs? we should not allow those ...

Reply:
For this I'll create an improvement